chapters/02-ruby_on_rails/readme.md

```diff
@@ -0,0 +1,795 @@
+special tags `<%= %>`.
+
+This will render Ruby code we put inside of it. So this code should show how
+many dogs do you really want to have. Refresh your browser to see the result.
+
+```html
+<!-- app/views/landing/index.html.erb -->
+<h1>Landing#index</h1>
+<p>Find me in app/views/landing/index.html.erb</p>
+
+<div id="dog-div">
+    <p class="grow-up">
+        When I grow up I'd like to have at least <%= @result %>
+        <%= @dog_breed %>s
+    </p>
+</div>
+```
+
+Other neat trick that you will now be able to do is to display generate multiple
+page elements like in a loop. You can surround ruby control logic with `<% %>`
+to execute it. Here, we will print what their names would be in an unordered
+list `<ul>`. For the number of times we will repeat the same block of code
+containing a single list item `<li>`. Take a not that we can also use local
+variables like *i* here and execute simple operations.
+
+```html
+<!-- app/views/landing/index.html.erb -->
+<!-- put this code inside of previous div (dog-div) -->
+<ul>
+    <% @result.times do |i| %>
+        <li>Vilko <%= i + 1 %>.</li>
+    <% end %>
+</ul>
+```
+
+The same pattern goes for writing if then statements. You just have to annotate
+`<% if <expression> %>` for the beginning of the block. You can use
+`<% elsif <expression> %>` to match multiple criteria and `<% else %>` to catch
+the rest of the cases. Just don't forget to add `<% end %>` to the end of the
+block. In this example dog breed is alphabetically compared to some reference
+string. This means that Ruby will compare two strings letter by letter. For
+example *Irish setter* is smaller than *Labrador* and *Maltese* is larger
+because of alphabetical order of the first letter. If first two letters were the
+same next ones would be compared.
+
+```html
+<!-- app/views/landing/index.html.erb -->
+<!-- put this code inside of previous div (dog-div) -->
+<% if @dog_breed > 'Labrador' %>
+    <p><%= @dog_breed %> is alphabetically a large dog</p>
+<% else %>
+    <p><%= @dog_breed %> is alphabetically a small dog</p>
+<% end %>
+```
+
+You will again have to refresh the page to show the new results.
+
+![Using data from controller in the view](images/rails-index-view.png)
+
+*If it works, it might not be a bad idea to commit these changes.*
+
+### Making it pretty
+
+Using controllers and views you can display many thing and control how the
+application works. But as you can see this web doesn't really look that nice.
+Maybe we could fix that a bit?
+
+In this chapter we will use a few of these front-end technologies but they will
+be more thoroughly explained in chapter 5. So use resources and guides regarding
+css and javascript as is.
+
+[Bootstrap](http://getbootstrap.com) is an easy to use framework which will help
+us make pretty web pages without much effort! To install it we can use gem. Open
+`Gemfile` in the root of your project and add `gem 'bootstrap'` to include it to
```

Comment:
If I'm not mistaken, you need **bootstrap-sass** gem, not just bootstrap

Reply:
bootstrap-sass is for bs 3 while bootstrap is used for v4